### PR TITLE
#27500 Prepare Payment module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by slavvka** [comment]
```
18:23:49 PHP Fatal error:  Declaration of Magento\CustomerBalance\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp() must be compatible with Magento\Payment\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp(): void in /var/www/html/app/code/Magento/CustomerBalance/Test/Unit/Model/Cart/SalesModel/OrderTest.php on line 8
```

File: app/code/Magento/Payment/Test/Unit/Helper/DataTest.php [hunk 1]
```diff
@@ -1,48 +1,58 @@
...
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
 use Magento\Framework\TestFramework\Unit\Matcher\MethodInvokedAtIndex;
+use Magento\Framework\View\Element\BlockInterface;
+use Magento\Framework\View\LayoutInterface;
+use Magento\Payment\Helper\Data;
+use Magento\Payment\Model\Info;
+use Magento\Payment\Model\Method\AbstractMethod;
+use Magento\Payment\Model\MethodInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+
```
Comment (ihor-sviziev) [inline app/code/Magento/Payment/Test/Unit/Helper/DataTest.php:21]: Could you remove this line?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz,
Could you fix the small issue that I found on review and fatal error that @slavvka wrote above?

**Review by slavvka — CHANGES_REQUESTED** [review]
The same inheritance issue:
```
02:43:31 PHP Fatal error:  Declaration of Magento\CustomerBalance\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp() must be compatible with Magento\Payment\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp(): void in /var/www/html/app/code/Magento/CustomerBalance/Test/Unit/Model/Cart/SalesModel/OrderTest.php on line 8
```

**Comment by slavvka** [comment]
Unit EE, B2B:
```
15:41:16 PHP Fatal error:  Declaration of Magento\CustomerBalance\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp() must be compatible with Magento\Payment\Test\Unit\Model\Cart\SalesModel\OrderTest::setUp(): void in /var/www/html/app/code/Magento/CustomerBalance/Test/Unit/Model/Cart/SalesModel/OrderTest.php on line 8
```

**Comment by slavvka** [comment]
@magento run all tests

**Review by slavvka — APPROVED** [review]

**Review by slavvka — APPROVED** [review]

**Pull request merged**